R/cql-geom-predicates.R

```diff
@@ -41,11 +41,17 @@ bcdc_cql_string <- function(x, geometry_predicates, pattern = NULL,
                             distance = NULL, units = NULL,
                             coords = NULL, crs = NULL){
 
+  if (inherits(x, "sql")) {
+    stop(glue::glue("object {as.character(x)} not found.\n The object passed to {geometry_predicates} needs to be valid sf object."),
+         call. = FALSE)
+  }
+
```

Comment:
I wonder if this should be below the `bcdc_promise` check (i.e., line 53) and be more like:

```
if (!inherits(x, c("sf", "sfc", "sfg"))) {
  stop(glue::glue("object '{deparse(substitute(x))}' not found.\n The object passed to {geometry_predicates} needs to be valid sf object."),
       call. = FALSE)
}
```

Reply:
Yep I am fine with that. The rationale with using the `sql` class is that it would never be in that function if a user wasn't intending to pass a spatial object into it so it is reasonably safe. The reason I want to test for that class is because here the `deparse(substitute` solution won't work a x doesn't actually exist as an object in the environment. That's why the `as.character` was used.

Reply:
Oh I see - that makes sense! Ignore my comment then, I think you have it right. Leave it as is, but maybe add a comment in the code explaining it a bit? 

Also, my solution would bonk when this is called from `BBOX`, because `x = NULL` in that case.